packages/dev/gui/src/2D/controls/image.ts

```diff
@@ -620,7 +620,7 @@ export class Image extends Control {
      * @returns the svg
      */
     private _svgCheck(value: string): string {
-        if (window.SVGSVGElement && value.search(/.svg#/gi) !== -1 && value.indexOf("#") === value.lastIndexOf("#")) {
+        if (window.SVGSVGElement && value.search(/.svg.*#/gi) !== -1 && value.indexOf("#") === value.lastIndexOf("#")) {

```

Comment:
should probably be:

```suggestion
        if (window.SVGSVGElement && value.search(/\.svg.*#/gi) !== -1 && value.indexOf("#") === value.lastIndexOf("#")) {
```
Because the change will make any file with svg anywhere in the filename to work.
At least with this page there will have to be `.svg` in the filename. A better suggestion should be an optional `?` as part of the regexp

Reply:
You are right, adding the `\` should narrow a bit the matching.

Also, yes, probably a `?.*` and a `#.*` optional cases might narrow even more the case i want to match.

Reply:
How about this?

`(\.svg|\.svg?[?|#].*)$`

It should match:

```
file.svg
file.svg#someId
file.svg?param1=1
file.svg?param1=1&param2=2
file.svg?param1=1&param2=2#someId
```

but not:

```
file.svgAndSomeOtherStuff
```

Of course, even with this check, you can still write whatever you want after a `?` or `#`, also i don't think that this rule should sanitize a given url, doing that will over complicate this regex maybe.